src/stats.rs

```diff
@@ -20,15 +27,39 @@ static LATEST_STATS: Lazy<Mutex<HashMap<usize, HashMap<String, i64>>>> =
 /// 15 seconds.
 static STAT_PERIOD: u64 = 15000;
 
+/// The various states that a client can be in
+#[derive(Debug, Clone, Copy)]
+pub enum ClientState {
+    ClientWaiting,
+    ClientActive,
+    ClientIdle,
+}
+
+/// Information we keep track off which can be queried by SHOW CLIENTS
+#[derive(Debug, Clone)]
+pub struct ClientInformation {
+    pub state: ClientState,
+    pub process_id: i32,
+
+    pub application_name: String,
+    pub username: String,
+    pub pool_name: String,
+
+    pub transaction_count: u64,
+    pub query_count: u64,
+    pub error_count: u64,
+}
+
 /// The names for the events reported
 /// to the statistics collector.
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone)]
 enum EventName {
     CheckoutTime,
     Query,
     Transaction,
     DataSent,
     DataReceived,
+    ClientRegistered(String, String, String),
```

Comment:
This is really cool

Reply:
Let's make this either a struct or at least named parameters

Reply:
I already made a change to my local branch to use anonymous struct